### PR TITLE
[ACR] New API version 2020-11-01-preview: AZ and On-Prem support

### PR DESCRIPTION
**Pull request by jaysterp**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2020-11-01-preview/containerregistry.json [hunk 1]
```diff
@@ -0,0 +1,5633 @@
...
+          "items": {
+            "$ref": "#/definitions/StatusDetailProperties"
+          },
+          "readOnly": true
+        }
+      }
+    },
+    "ParentProperties": {
+      "description": "The properties of the connected registry parent.",
+      "required": [
```
Comment (adewaleo) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2020-11-01-preview/containerregistry.json:2663]: should id property be added here @jaysterp, its being generated locally

Comment (jaysterp) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2020-11-01-preview/containerregistry.json:2663]: It is being removed from dev.

**Comment by rosanch** [comment]
@jaysterp FYI  you need to update the readme file

**Comment by majastrz** [comment]
>     "TokenProperties": {

Are any of the token APIs returning any secret values anywhere? 

---
Refers to: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2020-11-01-preview/containerregistry.json:4693 in d6323fc. [](commit_id = d6323fcf6ff5099650c92a44a00fe5a81913d25e, deletion_comment = False)

**Comment by majastrz** [comment]
>     "TokenProperties": {

I don't see anything like that. If you do, make sure not to return the secret value on GETs and annotate the property with x-ms-secret: true.

---
In reply to: [719731622](https://github.com/Azure/azure-rest-api-specs/pull/11412#issuecomment-719731622) [](ancestors = 719731622)

---
Refers to: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2020-11-01-preview/containerregistry.json:4693 in d6323fc. [](commit_id = d6323fcf6ff5099650c92a44a00fe5a81913d25e, deletion_comment = False)

**Review by majastrz — APPROVED** [review]
Reviewed the delta with `2019-12-01-preview`. The new APIs look fine to me. Signed off from ARM side with a couple of comments:
- Take a look and my question about secrets. I think you're fine but let's double check.
- You have a lot of linter errors. You will need to fix them before merging.

**Comment by fengzhou-msft** [comment]
@jaysterp can you fix the CI errors following the [fix guide](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md)?

**Comment by pilor** [comment]
Changes look good.  ARM sign-off is still applicable

**Review by pilor — APPROVED** [review]

**Comment by jaysterp** [comment]
@fengzhou-msft the duplicate model collision for the `ProxyResource` definition will be resolved when this PR is merged: https://github.com/Azure/azure-rest-api-specs/pull/11601, which adds `SystemData` to the proxy resource definition in Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json.

**Comment by jaysterp** [comment]
@fengzhou-msft can you take a look at the model validation error for OBJECT_ADDITIONAL_PROPERTIES? `location` and `tags` are brought into the `registries`, `replications`, and `webhooks` resource definitions via `allOf` with `"$ref": "#/definitions/Resource"`. [Resource](https://github.com/Azure/azure-rest-api-specs/blob/fcfd66cfbb96dfa995a2c2e65bbff7d32a3bcd23/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2020-11-01-preview/containerregistry.json#L5545) includes these properties.

**Comment by jaysterp** [comment]
> > ```
> > "TokenProperties": {
> > ```
> 
> Are any of the token APIs returning any secret values anywhere?
> 
> Refers to: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2020-11-01-preview/containerregistry.json:4693 in [d6323fc](https://github.com/Azure/azure-rest-api-specs/commit/d6323fcf6ff5099650c92a44a00fe5a81913d25e). [](commit_id = [d6323fc](https://github.com/Azure/azure-rest-api-specs/commit/d6323fcf6ff5099650c92a44a00fe5a81913d25e), deletion_comment = False)

@majastrz The only time token credentials are available to the user is when they run the `registries/generateCredential` API, which is linked to a token resource. This is a one-time return of their token password(s). The passwords can't be retrieved again, but new ones can be generated through the same API. Secret values are not exposed through any of the other token APIs.

**Comment by jaysterp** [comment]
> @fengzhou-msft can you take a look at the model validation error for OBJECT_ADDITIONAL_PROPERTIES? `location` and `tags` are brought into the `registries`, `replications`, and `webhooks` resource definitions via `allOf` with `"$ref": "#/definitions/Resource"`. [Resource](https://github.com/Azure/azure-rest-api-specs/blob/fcfd66cfbb96dfa995a2c2e65bbff7d32a3bcd23/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2020-11-01-preview/containerregistry.json#L5545) includes these properties.

This model validation error was fixed upon adding a local definition of SystemData to the `Resource` definition. cc @hkuadithya

**Comment by jaysterp** [comment]
@fengzhou-msft is there an issue with the GO SDK test?

**Comment by fengzhou-msft** [comment]
> @fengzhou-msft is there an issue with the GO SDK test?

@ArcturusZhang can you take a look at the Go SDK generation for ACR? Anything wrong in configuration?

**Comment by ArcturusZhang** [comment]
This is a pre-existing failure, I will open a PR to fix this - please hold on merging this PR until the fix is merged.

**Comment by ArcturusZhang** [comment]
Hi @fengzhou-msft could you please have a look at this PR? https://github.com/Azure/azure-rest-api-specs/pull/11781

And @jaysterp could you please update your PR after the above fix PR is merged?

**Comment by ArcturusZhang** [comment]
Hi @jaysterp my fix has been merged, could you please update your code by merging the master branch in?

**Comment by jaysterp** [comment]
Thank you @ArcturusZhang. This resolved the issue. @fengzhou-msft we are ready to merge.

**Pull request merged**